test/services/keyVault/keyVault-key-tests.js

```diff
@@ -0,0 +1,658 @@
+          next();
+        });
+      }
+
+      function updateKey(keyUri, next) {
+        var updatingBundle = KvUtils.clone(createdBundle);
+        updatingBundle.attributes.exp = new Date('2050-02-02T08:00:00.000Z');
+        updatingBundle.key.key_ops = ['encrypt', 'decrypt'];
+        updatingBundle.tags = { foo: random.hex(100) };
+        var request = { key_ops: updatingBundle.key.key_ops, attributes: updatingBundle.attributes, tags: updatingBundle.tags };
+        client.updateKey(keyUri, request, function(err, keyBundle) {
+          if (err) throw err;
+          updatingBundle.attributes.updated = keyBundle.attributes.updated;
+          compareObjects(updatingBundle, keyBundle);
+          createdBundle = keyBundle;
+          next();
+        });
+      }
+      
+      function updateKeyWOVersion(next) {
+        return updateKey(keyId.baseIdentifier, next);
+      }
+      
+      function updateKeyWithVersion(next) {
+        return updateKey(keyId.identifier, next);
+      }
+
+      function deleteKey(next) {
+        client.deleteKey(keyId.vault, keyId.name, function(err, keyBundle) {
+          if (err) throw err;
+          compareObjects(createdBundle, keyBundle);
+          next();
+        });
+      }
+
+      function getKeyReturnsNotFound(next) {
+        client.getKey(keyId.baseIdentifier, function(err, keyBundle) {
+          if (!err || !err.code || err.code !== 'KeyNotFound' || !err.statusCode || err.statusCode !== 404) throw new Error('Unexpected error object: ' + JSON.stringify(err, null, ' '));
+          next();
+        });
+      }
+
+      series([
+        createKey,
+        getKeyWOVersion,
+        getKeyWithVersion,
+        updateKeyWOVersion,
+        updateKeyWithVersion,
+        deleteKey,
+        getKeyReturnsNotFound,
+        function () {done();}
+      ]);     
+      
+    });
+  });
+  
+  describe('import', function() {
+    it('should work', function(done) {
+      
+      function doImport(importToHardware, next) {
+        var importKeyRequest = {
+          key: {
+            kty: 'RSA',
+            key_ops: ['encrypt', 'decrypt', 'sign', 'verify', 'wrapKey', 'unwrapKey']
+          },
+          hsm: importToHardware
+        };  
+        setRsaParameters(importKeyRequest.key, getTestKey(suiteUtil));
+        client.importKey(vaultUri, KEY_NAME, importKeyRequest, function(err, keyBundle) {
+          if (err) throw err;
+          validateRsaKeyBundle(keyBundle, vaultUri, KEY_NAME, importToHardware ? 'RSA-HSM' : 'RSA', importKeyRequest.key_ops);
+          next();
+        });
+      };
+      
+      function importToSoftware(next) {
+        doImport(false, next);        
+      }
+            
+      function importToHardware(next) {
+        doImport(false, next);        
+      }
+
+      series([
+        importToSoftware,
+        importToHardware,
+        function() {done();}
+      ]);     
+
+    });
+  });
+ 
+  xdescribe('list', function() {
```

Comment:
as i mentioned later, please use ".skip". This coed will coexist with many other codes, it is important to be consistent